testsuite/tests/typing-jkind-bounds/subsumption/functors.ml

```diff
@@ -0,0 +1,95 @@
+(* TEST
+    flags = "-extension layouts_alpha";
+    expect;
+*)
+
+module F (M : sig type t end) = struct
+  type t : immutable_data with M.t
+end
+
+module Int = struct
+  type t = int
+end
+type t : immutable_data = F(Int).t
+[%%expect {|
+module F :
+  functor (M : sig type t end) -> sig type t : immutable_data with M.t end
+module Int : sig type t = int end
+type t = F(Int).t
+|}]
+
+module T = struct
+  type t
+end
+type t : immutable_data = F(T).t
+[%%expect {|
+module T : sig type t end
+Line 4, characters 0-32:
+4 | type t : immutable_data = F(T).t
+    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Error: The kind of type "F(T).t" is immutable_data with T.t
+         because of the definition of t at line 2, characters 2-34.
+       But the kind of type "F(T).t" must be a subkind of immutable_data
+         because of the definition of t at line 4, characters 0-32.
+|}]
+
+module F (M : sig type 'a t end) = struct
+  type 'a t : immutable_data with 'a M.t
+end
+
+module Ref = struct
+  type 'a t = 'a ref
+end
+type 'a t : mutable_data with 'a = 'a F(Ref).t
```

Comment:
Maybe try this test with `immutable_data`, which should fail. And also missing the `with 'a`, which should also fail.